rflx/integration.py

```diff
@@ -0,0 +1,173 @@
+from pydantic.types import ConstrainedInt
+
+from rflx.error import Location, RecordFluxError, Severity, Subsystem
+from rflx.identifier import ID
+from rflx.model import Model
+from rflx.model.session import Session
+
+
+class IntSize(ConstrainedInt):
+    gt = 0
+
+
+class SessionSize(BaseModel):
+    default: Optional[IntSize] = Field(alias="Default")
+    global_: Optional[Dict[str, IntSize]] = Field(alias="Global")
+    local_: Optional[Dict[str, Dict[str, IntSize]]] = Field(alias="Local")
+
+
+class SessionIntegration(BaseModel):
+    buffer_size: SessionSize = Field(alias="Buffer_Size")
+
+
+class IntegrationFile(BaseModel):
+    session: Dict[str, SessionIntegration] = Field(alias="Session")
+
+
+class Integration:
+    @property
+    def defaultsize(self) -> int:
+        return 4096
+
+    def __init__(self) -> None:
+        self._packages: Dict[str, IntegrationFile] = {}
+
+    def add_integration_file(self, filename: Path, file: object, error: RecordFluxError) -> None:
+        try:
+            print(filename)
+            print(filename.stem)
+            self._packages[filename.stem] = IntegrationFile.parse_obj(file)
+        except ValidationError as e:
+            error.extend(
+                [(f"{e}", Subsystem.PARSER, Severity.ERROR, self._to_location(filename.stem))]
+            )
+
+    def validate(self, model: Model, error: RecordFluxError) -> None:
+        for package, integration_file in self._packages.items():
+            for session_name, integration in integration_file.session.items():
+                matching_sessions = [
+                    s
+                    for s in model.sessions
+                    if package == str(s.package).lower() and str(s.identifier.name) == session_name
+                ]
+                if not matching_sessions:
+                    error.extend(
+                        [
+                            (
+                                f'unknown session "{session_name}"',
+                                Subsystem.PARSER,
+                                Severity.ERROR,
+                                Integration._to_location(package),
+                            )
+                        ]
+                    )
+                    return
+                assert len(matching_sessions) == 1
+                session = matching_sessions[0]
+                self._validate_globals(package, integration, session, error)
+                self._validate_states(package, integration, session, error)
+
+    def get_size(self, session: ID, variable: ID, state: Optional[ID]) -> int:
+        """
+        Return the requested buffer size for a variable of a given session and state.
+
+        If [state] is None, the variable is assumed to be a global
+        variable. If no specific buffer size was requested for the variable,
+        returns the default buffer size for the session, if present, or the
+        default buffer size for RecordFlux.
+        """
```

Comment:
The brackets around `state` seem confusing to me. This syntax is not used for any other parameter (neither in this docstring nor in any other).

I think it should be either just "If ..., return the ..." or "If ..., the function returns the ...".

Reply:
Fixed in d095c573a1defa433803ca5da716b75140cbca52 and 7ed4f5528401250ea059ec3e37e7c78852b72e61